tests/ludwig/automl/test_base_config.py

```diff
@@ -1,29 +1,45 @@
+import random
+
 import pytest
 
-from ludwig.automl.base_config import should_exclude, infer_type
+from ludwig.automl.base_config import infer_type, should_exclude
 from ludwig.automl.utils import FieldInfo
-from ludwig.constants import TEXT, NUMERICAL, CATEGORY, BINARY, IMAGE
+from ludwig.constants import BINARY, CATEGORY, IMAGE, NUMERICAL, TEXT
+from ludwig.data.dataset_synthesizer import generate_string
 
 ROW_COUNT = 100
 TARGET_NAME = 'target'
 
 
-@pytest.mark.parametrize("num_distinct_values,distinct_values,avg_words,img_values,missing_vals,expected", [
-    (ROW_COUNT, ['1.1','2.2'], 0, 0, 0.0, NUMERICAL),
-    (10, ['0','1','2','3','4','5','6','7','8','9'], 0, 0, 0.0, CATEGORY),
-    (2, ['0', '1'], 0, 0, 0.0, BINARY),
-    (2, ['human', 'bot'], 0, 0, 0.0, CATEGORY),
-    (2, ['1.5', '3.7'], 0, 0, 0.1, NUMERICAL),
-    (ROW_COUNT, [], 3, 0, 0.0, TEXT),
-    (ROW_COUNT, [], 1, ROW_COUNT, 0.0, IMAGE),
-])
-def test_infer_type(num_distinct_values, distinct_values, avg_words, img_values, missing_vals, expected):
+@pytest.mark.parametrize(
+    "num_distinct_values,distinct_values,img_values,missing_vals,expected",
+    [
+        # Random numbers.
+        (ROW_COUNT, [str(random.random()) for _ in range(ROW_COUNT)], 0, 0.0, NUMERICAL),
+        # Random numbers with NaNs.
+        (ROW_COUNT, [str(random.random()) for _ in range(ROW_COUNT - 1)] + ['NaN'], 0, 0.0,
+         NUMERICAL),
+        # Finite list of numbers.
+        (10, ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'], 0, 0.0, CATEGORY),
+        (2, ['1.5', '3.7'], 0, 0.1, CATEGORY),
```

Comment:
This would be an odd dataset, but in this case the entire feature column consists of 2 unique numerical values. Would that still be a regression problem?

Reply:
Well, the ultimate responsibility for that call would be user intent, and we allow the user to override our selections.
FWIW, my personal take on selection behavior when user intent is unknown, is if a column is using floating point,
it suggests magnitude is relevant, and even if the training data only shows a few (like 2) different values,
there is relative information there.

Reply:
Discussed offline! We'll continuing labeling this case as numerical with the assumption that even with limited values, the continuity semantics are important.